TripalCultivate/TripalCultivate
# G1.04 Study Experiment Fields

This issue expands on issue #4 

## What does this PR do?

Updated all the attribution fields by making changes to both field type and storage information. Then create multiple new Chado Contact, followed by creating experiments and assigning contacts to each role.

## Testing

1. Create a fresh docker on this branch:
```

docker build --build-arg drupalversion=10.2.x-dev --build-arg phpversion=8.3 --tag=trpcultivate-base:18-experimentFields ./
docker run --publish=80:80 -tid --name=base18 --volume=$(pwd):/var/www/drupal/web/modules/contrib/TripalCultivate trpcultivate-base:18-experimentFields
docker exec base18 service postgresql restart
```
2. Go to Tripal Admin>Content >Add Tripal Content>General>Contact to create multiple contacts

3. Then create one or multiple experiment(s) by going to Tripal Admin>Content >Add Tripal Content> Experiment Types

4. Under Content, now you should be able to find the Contacts and Experiments you've created

5. Go to one or multiple of your created experiments, click edit, then to see if you can find the contacts you just created in the dropdown, and you should be able to assign one or multiple contacts to each one of the attribution fields.

## Review

I have decided to leave the temporary changes that allow us to test this PR in place. Since the SQL is already committed in this branch, it doesn't make much difference to leave it there and remove when Tripal core is ready as compared to removing it now.

**So you can use the testing instructions above.**

That said, you can also just check out these screenshots and look at the tripalfield YAML file in this PR. Each block in this YAML file refers to a single field being added to a content type. The main parts to check are the label, description and term. The term information is in the terms YAML. The description is meant to be helpful text to the researcher editing the page to tell them how to fill it in.

### Tripal Content Type listing
<img width="1391" alt="Screenshot 2024-07-04 at 9 45 55 AM" src="https://github.com/TripalCultivate/TripalCultivate/assets/1566301/8a409fa4-7fe5-4e45-94f8-e80af10b5aef">

### Biochemical Assay

#### Manage Fields
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/fd54abd5-3fca-4cbf-a3c3-59f98528746e)

#### Add Content
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/67a5742e-f95a-4170-b913-65c704cdc700)

### Field Experiment

#### Manage Fields
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/eed5dbbc-273a-402a-9ab2-54dccb64e181)

#### Add Content
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/25cb2bc7-4b2e-450c-879b-a92425240839)

### Greenhouse Experiment

#### Manage Fields
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/a16656a9-9d4f-4406-8f1d-18eadf80c5fa)

#### Add Content
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/73113848-6844-44fa-8204-d2dccca0f8cc)

### Growth Chamber Experiment

#### Manage Fields
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/a2287981-842d-4d3f-8ba8-78148da74c86)

#### Add Content
![image](https://github.com/TripalCultivate/TripalCultivate/assets/1566301/bf732b85-9f24-45a9-ba04-8c1928482945)
